js_modules/dagit/packages/core/src/ui/useFormatDateTime.test.tsx

```diff
@@ -0,0 +1,134 @@
+import {render, screen} from '@testing-library/react';
+import * as React from 'react';
+
+import {TimezoneProvider, TimezoneStorageKey} from '../app/time/TimezoneContext';
+
+import {useFormatDateTime} from './useFormatDateTime';
+
+jest.mock('../app/time/browserTimezone', () => ({
+  browserTimezone: jest.fn(() => 'Pacific/Honolulu'),
+}));
+
+const NYD_2023_CST = 1672552800000;
+
+describe('useFormatDateTime', () => {
+  interface Props {
+    date: Date;
+    options: Intl.DateTimeFormatOptions;
+  }
+
+  const Test: React.FC<Props> = ({date, options}) => {
+    const formatDateTime = useFormatDateTime();
+    return <div>{formatDateTime(date, options)}</div>;
+  };
+
+  afterEach(() => {
+    window.localStorage.removeItem(TimezoneStorageKey);
+  });
+
+  describe('Automatic timezone', () => {
+    // Timezone has not been set in localStorage, so it should be an `Automatic` timezone, and
```

Comment:
Why do we set the Timezone in localStorage? I'd think we should just always use automatic?